packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -326,48 +273,59 @@ export function useGridEditRows(
     [apiRef],
   );
 
-  const handleDoubleClick = React.useCallback(
+  const handleCellEditExit = React.useCallback(
+    (params: GridCellParams, event?: React.SyntheticEvent) => {
+      // TODO check if its propagation was stopped
+      setCellMode(params.id, params.field, 'view');
+
+      // When dispatched by the document, the event is not passed
+      if (!event || !isKeyboardEvent(event)) {
+        return;
+      }
+
+      if (isCellEditCommitKeys(event.key)) {
+        apiRef.current.publishEvent(GRID_CELL_NAVIGATION_KEY_DOWN, params, event);
+        return;
+      }
+      if (event.key === 'Escape' || isDeleteKeys(event.key)) {
+        apiRef.current.setCellFocus(params.id, params.field);
+      }
+    },
+    [apiRef, setCellMode],
+  );
+
+  const handleCellDoubleClick = React.useCallback(
     (...args) => {
+      // TODO don't publish if cell is not editable
       apiRef.current.publishEvent(GRID_CELL_EDIT_ENTER, ...args);
     },
     [apiRef],
   );
 
   useGridApiEventHandler(apiRef, GRID_CELL_KEY_DOWN, handleCellKeyDown);
   useGridApiEventHandler(apiRef, GRID_CELL_MOUSE_DOWN, preventTextSelection);
-  useGridApiEventHandler(apiRef, GRID_CELL_DOUBLE_CLICK, handleDoubleClick);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_ENTER, handleEnterEdit);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_EXIT, handleExitEdit);
+  useGridApiEventHandler(apiRef, GRID_CELL_DOUBLE_CLICK, handleCellDoubleClick);
   useGridApiEventHandler(apiRef, GRID_CELL_FOCUS_OUT, handleCellFocusOut);
-  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_PROPS_CHANGE, handleCellEditPropsChange);
   useGridApiEventHandler(apiRef, GRID_COLUMN_HEADER_DRAG_START, handleColumnHeaderDragStart);
-  useGridApiEventHandler(
-    apiRef,
-    GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
-    handleCellEditPropsChangeCommited,
-  );
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_ENTER, handleCellEditEnter);
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_EXIT, handleCellEditExit);
+  useGridApiEventHandler(apiRef, GRID_CELL_EDIT_COMMIT, handleCellEditCommit);
+  useGridApiEventHandler(apiRef, GRID_EDIT_CELL_PROPS_CHANGE, handleEditCellPropsChange);
 
-  useGridApiOptionHandler(
-    apiRef,
-    GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
-    props.onEditCellChangeCommitted,
-  );
-  // TODO rename to onEditCellPropsChange
-  useGridApiOptionHandler(apiRef, GRID_CELL_EDIT_PROPS_CHANGE, props.onEditCellChange);
-  // TODO remove because onEditCellChange does the same
-  useGridApiOptionHandler(apiRef, GRID_CELL_VALUE_CHANGE, props.onCellValueChange);
-  useGridApiOptionHandler(apiRef, GRID_CELL_MODE_CHANGE, props.onCellModeChange);
-  useGridApiOptionHandler(apiRef, GRID_ROW_EDIT_MODEL_CHANGE, props.onEditRowModelChange);
+  useGridApiOptionHandler(apiRef, GRID_CELL_EDIT_COMMIT, options.onCellEditCommit);
+  // TODO remove, use onEditRowsModelChange directly
```

Comment:
If we are going in this direction. Should we update the docs now to not use `onEditRowsModelChange` or do it later?

Reply:
> Should we update the docs now to not use onEditRowsModelChange or do it later?

If you meant `onEditCellPropsChange` I already updated all markdown pages and demos to use `onEditRowsModelChange`.

Reply:
I'm referring to `onEditCellPropsChange` yes. We can see in https://master--material-ui-x.netlify.app/components/data-grid/editing/#server-side-validation that we document the prop. Is it expected?

Reply:
Hmmm, maybe we can't remove `onEditCellPropsChange` yet. To do server-side validation, the user will want to make the server request only when a particular field is changed. With `onEditRowsModelChange` we loose the field that caused the change.

Reply:
Then it's not clear to me. Why is the value not the only prop we care about for the edit cell?

Reply:
To be clear. When I say "the props of a field changes" I'm meaning the props in the `editRows` state.

<img width="350" src="https://user-images.githubusercontent.com/42154031/126718927-c393c771-c277-4034-a9ea-c970062564e4.png" alt="image" style="max-width:100%;">

Going back to https://github.com/mui-org/material-ui-x/pull/2143#discussion_r675085418. If we use `onEditRowsModelChange` for validation, a server request might be made even if the field was not changed. This method will be shared with the row editing, so multiple fields could be being modified at the same time. Imagine an user registration form. I only want to validate the username when the user changes the username field, not when the name or password is changed. I can only have this granularity with `onEditCellPropsChange`.

Reply:
Ok, I see, the alternative would be to add a new argument to `onEditRowsModelChange`. We have often solve this type of problem like this in the past